### PR TITLE
chore(monitor): add info log for client dns resolve

### PR DESCRIPTION
**Pull request by MrCroxx**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Add rpc client dns resolve info log.

#14207 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/src/monitor/connection.rs [hunk 1]
```diff
@@ -388,6 +391,92 @@ pub fn monitor_connector<C>(
...
+}
+
+impl Future for MonitoredGaiFuture {
+    type Output = Result<MonitoredGaiAddrs, Error>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        Pin::new(&mut self.inner).poll(cx).map(|res| match res {
+            Ok(addrs) => {
+                let addrs: MonitoredGaiAddrs = addrs.into();
+                info!("resolve {} => {:?}", self.name, addrs.inner);
```
Comment (arkbriar) [inline src/common/src/monitor/connection.rs:436]: I think it's more appropriate to log in DEBUG level.

**Review by arkbriar — APPROVED** [review]
LGTM!

**Review by lmatz — APPROVED** [review]
LGTM, a wrapper of https://docs.rs/hyper/0.14.1/src/hyper/client/connect/dns.rs.html#45-57

**Review by Li0k — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**